internal/servers/worker.go

```diff
@@ -41,7 +48,142 @@ func (w *Worker) clone() *Worker {
+func (w *Worker) CanonicalTags() map[string][]string {
+	dedupedTags := make(map[Tag]struct{})
+	for _, t := range w.Tags {
+		dedupedTags[*t] = struct{}{}
+	}
+	if w.ReportedStatus != nil {
+		for _, t := range w.ReportedStatus.Tags {
+			dedupedTags[*t] = struct{}{}
+		}
+	}
+	tags := make(map[string][]string)
+	for t := range dedupedTags {
+		tags[t.Key] = append(tags[t.Key], t.Value)
+	}
+	return tags
+}
+
+// LastConnectionUpdate contains the last time the worker has reported to the
+// controller its connection status.  If the worker has never reported to a
+// controller then nil is returned.
+func (w *Worker) LastConnectionUpdate() *timestamp.Timestamp {
+	if w.ReportedStatus == nil {
+		return nil
+	}
+	return w.ReportedStatus.GetUpdateTime()
+}
+
 // TableName overrides the table name used by Worker to `server_worker`
 func (Worker) TableName() string {
 	return "server_worker"
 }
+
+// workerAggregate contains an aggregated view of the values associated with
+// a single worker.
+type workerAggregate struct {
+	PublicId               string `gorm:"primary_key"`
+	ScopeId                string
+	Name                   string
+	Description            string
+	CreateTime             *timestamp.Timestamp
+	UpdateTime             *timestamp.Timestamp
+	Address                string
+	Version                uint32
+	ApiTags                string
+	WorkerStatusName       string
+	WorkerStatusAddress    string
+	WorkerStatusCreateTime *timestamp.Timestamp
+	WorkerStatusUpdateTime *timestamp.Timestamp
+	WorkerConfigTags       string
+}
+
+func (a *workerAggregate) toWorker(ctx context.Context) (*Worker, error) {
+	const op = "servers.(workerAggregate).toWorker"
+	workerOptions := []Option{
+		WithPublicId(a.PublicId),
+		WithName(a.Name),
+		WithDescription(a.Description),
+		WithAddress(a.Address),
+	}
+	tags, err := tagsFromAggregatedTagString(ctx, a.ApiTags)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("error parsing config tag string"))
+	}
+	if len(tags) > 0 {
+		workerOptions = append(workerOptions, WithWorkerTags(tags...))
+	}
+	worker := NewWorker(a.ScopeId, workerOptions...)
+	worker.CreateTime = a.CreateTime
+	worker.UpdateTime = a.UpdateTime
+	worker.Version = a.Version
+
+	if a.WorkerStatusCreateTime == nil {
+		return worker, nil
+	}
+	statusOptions := []Option{
+		WithName(a.WorkerStatusName),
+		WithAddress(a.WorkerStatusAddress),
+	}
+	tags, err = tagsFromAggregatedTagString(ctx, a.WorkerConfigTags)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("error parsing config tag string"))
+	}
+	if len(tags) > 0 {
+		statusOptions = append(statusOptions, WithWorkerTags(tags...))
+	}
+	cfg := NewWorkerStatus(a.PublicId, statusOptions...)
+	cfg.CreateTime = a.WorkerStatusCreateTime
+	cfg.UpdateTime = a.WorkerStatusUpdateTime
+
+	worker.ReportedStatus = cfg
+	return worker, nil
+}
+
+func tagsFromAggregatedTagString(ctx context.Context, s string) ([]*Tag, error) {
```

Comment:
Wondering if this function should be documented to explain the delimiters/ parsing.